### PR TITLE
Fix various typos

### PR DESCRIPTION
**Pull request by luzpaz**
Found via `codespell -q 3 -S ./src/cairo,./src/dxflib,./src/clipper,tpl_lib,*.ts -L ba,pard,parm,trough`

Note some source documentation has been modified.

### CODE REVIEW

File: doc/research/tool_sweep/camotics_3_axis_cnc_simulation.tex [hunk 1]
```diff
@@ -73,7 +73,7 @@ \section{Docks \& Toolbars}
 \section{Icons}
 \section{Qt Creator}
 
-\chapter{Slice Optimzation}
+\chapter{Slice Optimization}
```
Comment (luzpaz) [inline doc/research/tool_sweep/camotics_3_axis_cnc_simulation.tex:76]: Please review

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "CAMotics",
+  "version": "1.3.0",
+  "lockfileVersion": 3,
```
Comment (luzpaz) [inline package-lock.json:4]: Not sure how this file was modified. Will revert

**Comment by jcoffland** [comment]
Looks good but please remove the package-lock.json file then squash.  I leave this file out intentionally.

File: package-lock.json (hunk 2 above)
Comment (luzpaz) [reply package-lock.json]: reverted

**Comment by luzpaz** [comment]
Done.

**Comment by jcoffland** [comment]
Thanks!

**Pull request merged**